src/main/kotlin/com/jaredsburrows/license/LicenseReportTask.kt

```diff
@@ -96,43 +96,24 @@ open class LicenseReportTask : DefaultTask() { // tasks can't be final
 
     // If Android project, add extra configurations
     variantName?.let { variant ->
-      // Add buildType configurations
-      configurations.find { it.name == "compile" }?.let {
-        configurationSet.add(configurations.getByName("${buildType}Compile"))
-      }
-      configurations.find { it.name == "api" }?.let {
-        configurationSet.add(configurations.getByName("${buildType}Api"))
-      }
-      configurations.find { it.name == "implementation" }?.let {
-        configurationSet.add(configurations.getByName("${buildType}Implementation"))
-      }
-
-      // Add productFlavors configurations
-      productFlavors.forEach { flavor ->
-        // Works for productFlavors and productFlavors with dimensions
-        if (variant.capitalize().contains(flavor.name.capitalize())) {
-          configurations.find { it.name == "compile" }?.let {
-            configurationSet.add(configurations.getByName("${flavor.name}Compile"))
-          }
-          configurations.find { it.name == "api" }?.let {
-            configurationSet.add(configurations.getByName("${flavor.name}Api"))
-          }
-          configurations.find { it.name == "implementation" }?.let {
-            configurationSet.add(configurations.getByName("${flavor.name}Implementation"))
-          }
-        }
+      configurations.find { it.name == "${variant}RuntimeClasspath" }?.also {
```

Comment:
This is much cleaner, we no longer need to check for any "special" flavors?

Reply:
Not sure what do you mean by "special" flavors. This is based on my observations that all the dependencies can be found in ${variant}RuntimeClasspath configuration, including subproject dependencies which were missing in the other ones. I've seen there are some tests for flavored builds and since they have passed I assumed we should be fine.

Actually I've seen Google is doing it bit differently in their [oss licenses plugin](https://github.com/google/play-services-plugins/tree/master/oss-licenses-plugin). Instead of specifying configuration that should be taken into account in license generation, they're taking all the configurations and excluding only some of them like the ones that cannot be resolved or tests. Not sure which approach is better though.

Reply:
Ok thanks. I did not know about that plugin.

Reply:
I tried that plugin but it has it's flaws. Eg it cannot only be applied to a project of type `com.android.application` but not when it's `com.android.library` which is my use case.